packages/sdk/src/adapters/multipeer/protocols/clientSync.ts

```diff
@@ -202,15 +202,31 @@ export class ClientSync extends Protocols.Protocol {
 	 */
 	public 'stage:create-animations' = () => {
 		// Send all cached interpolate-actor and create-animation messages.
-		this.client.session.actors.map(syncActor => this.createActorInterpolations(syncActor));
-		this.client.session.actors.map(syncActor => this.createActorAnimations(syncActor));
+		for (const syncActor of this.client.session.actors) {
+			this.createActorInterpolations(syncActor);
+			this.createActorAnimations(syncActor);
+		}
+		// send all managed create-animation messages
+		for (const message of this.client.session.animationCreators) {
+			if (message.payload.type === 'create-animation') {
+				super.sendMessage(message);
+			}
+		}
 	};
 
 	/**
 	 * @hidden
 	 * Driver for the `sync-animations` synchronization stage.
 	 */
-	public 'stage:sync-animations' = async () => {
+	public 'stage:sync-animations' = () => {
+		// sync new-style animations
+		for (const anim of this.client.session.animations) {
+			if (anim.update) {
+				super.sendMessage(anim.update);
+			}
+		}
+
+		// sync legacy animations
 		const authoritativeClient = this.client.session.authoritativeClient;
 		if (!authoritativeClient) {

```

Comment:
All code in this method after your additions above should be removed. This is the source of the "timed out waiting for sync-animations" bug.

Reply:
Agreed, but I can't until I've replaced interpolations. They rely on this code for sync still. All other animations use the new stuff though.

---
In reply to: [364937498](https://github.com/microsoft/mixed-reality-extension-sdk/pull/475#discussion_r364937498) [](ancestors = 364937498)